p2panda-rs/src/schema/system_schema.rs

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+use std::collections::BTreeMap;
+use std::convert::TryFrom;
+use std::str::FromStr;
+
+use crate::document::DocumentView;
+use crate::operation::OperationValue;
+
+use super::SystemSchemaError;
+
+#[derive(Clone, Debug, Copy, PartialEq)]
+#[allow(missing_docs)]
+pub enum FieldType {
+    Bool,
+    Int,
+    Float,
+    String,
+    Relation,
+}
```

Comment:
Interesting! This is exactly the different "reincarnations" of fields, going through the "schema", "operation", "document view" process ..!

Wondering if there is a nice way to express this somehow, don't think it needs to be complicated (like nested enums or something), just maybe moving them closer to the "OperationValue` definitions (move them both into a new file?) and giving them different names like `OperationValueSchema`?

This is just a random thought, maybe it doesn't make sense at all .. I'll think more about it :thinking:

Reply:
Yeh, I think there could be a nice way of expressing this relationship. Let's keep it in mind :+1: